node_modules/oae-authentication/lib/api.js

```diff
@@ -677,6 +680,123 @@ var getUserIdFromLoginId = module.exports.getUserIdFromLoginId = function(tenant
 };
 
+
+/**
+ * Get the secret for an existing user's resetpassword request
+ *
+ * @param  {Context}    ctx             Standard context object containing the current user and the current tenant
+ * @param  {String}     tenantAlias     The tenant of the login id
+ * @param  {String}     userName        The user's own unique username 
+ * @param  {Function}   callback        Standard callback function
+ * @param  {Object}     callback.err    An error that occurred, if any
+ * @param  {Object}     callback.secret The secret token that will be returned
+ */
+var getResetPasswordSecret =module.exports.getResetPasswordSecret = function(ctx, tenantAlias, username, callback){
+	
+    // Associate the userID with loginID for later operations
+    tenantAlias = tenantAlias || ctx.tenant().alias;
+
+    // Create the loginid object first, so it can be passed into the validation
+    var loginId = new LoginId(tenantAlias, AuthenticationConstants.providers.LOCAL, username);
+	
+    // Generate a secret token
+    var secret = crypto.randomBytes(16).toString('hex');
+	
+    // Check if we can get an existing userID using the userName the user provided
+    _getUserIdFromLoginId(loginId, function(err, userId) {
+		
+        if (err) {
+	
+			return callback(err);
+        }
+		else{
+            // Store it to cassandra database with 24hr valid time USING CONSISTENCY QUORUM AND TTL 86400 
+            Cassandra.runQuery('UPDATE "AuthenticationLoginId" USING TTL 86400 SET "secret" = ? WHERE "loginId" = ?', [secret, _flattenLoginId(loginId)], function(err) {
+		        if (err) {
+		            log().error({'err': err}, 'Error in creating a secret token');
+		            return callback(err);
+		        }
+		        return callback();
+			});
+
+			// Get user's object as we will try to send the user an email with token
+			PrincipalsDAO.getPrincipal(userId, function(err, oldUser) {
+				if (err){
+					
+					return callback(err);
+				}
+				else{
+					
+					// Generate the email content with secret token
+					var emailData = {
+						'tenant': ctx.tenant(),
+						'user': oldUser,
+						'username': username,
+						'baseUrl': TenantsUtil.getBaseUrl(ctx.tenant()),
+						'skinVariables': require('oae-ui').getTenantSkinVariables(ctx.tenant().alias),
+						'secret': secret,
+					};
+
+					// Send the email to user(email address will probably be dispatched by sendEmail function)
+					EmailAPI.sendEmail('oae-authentication', 'reset', user, emailData, null, function(err) {
+				        if (err) {
+				            log().error({'err': err}, 'Error in sending an Email to the user');
+				            return callback(err);
+				        }
+				        return callback();
+					});
+				}
+			});
+		}
+	});
+	return callback(null,{'secret': secret});
+};
+
+/**
+ * Reset the password for an user with an existing username and a valid token
+ *
+ * @param    {Context}    ctx          Standard context object containing the current user and the current tenant
```

Comment:
You can probably add some more spacing between the variable names and their description as well (but make sure the columns line up for readability)